packages/next/shared/lib/router/router.ts

```diff
@@ -1389,6 +1389,12 @@ export default class Router implements BaseRouter {
       }
       Router.events.emit('routeChangeComplete', as, routeProps)
 
+      // A hash mark # is the optional last part of a URL
+      const hashRegex = /#.+$/
+      if (shouldScroll && hashRegex.test(as)) {
+        this.scrollToHash(as)
```

Comment:
Why is this extra `scrollToHash` call needed?

Reply:
When navigate to another page with non-latin anchors, the browser won't scroll to that anchor. We have to call `scrollToHash` manually after router change completed.